[WebDriver][BiDi] Use "javascript" as the "type" parameter when the message comes from a JavaScript error

lauromoura opened the pull request:
#### ede962eba432f83c2b61d7383105929804a22d6c
<pre>
[WebDriver][BiDi] Use &quot;javascript&quot; as the &quot;type&quot; parameter when the message comes from a JavaScript error
<a href="https://bugs.webkit.org/show_bug.cgi?id=285028">https://bugs.webkit.org/show_bug.cgi?id=285028</a>

Reviewed by Devin Rousso.

Use the `messageSource` field to check if the message is coming from a
JS error or from a regular console method.

Also set the `stackTrace` field for JS messages, as the spec requires
it.

* Source/WebDriver/Session.cpp:
(WebDriver::Session::doLogEntryAdded):
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::logEntryTypeForMessage):
(WebKit::WebAutomationSession::logEntryAdded):

Canonical link: <a href="https://commits.webkit.org/288666@main">https://commits.webkit.org/288666@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e0ef804f330ecb2fa27ccaa67d5148dccd93c11

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/84005 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/3623 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38306 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/89080 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/35013 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/3714 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/11591 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/65336 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23178 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87051 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/2766 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/76328 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45629 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/2697 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/30551 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/34062 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/73660 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31309 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90458 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11270 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/8158 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/73787 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/11494 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72157 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/73004 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18061 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17295 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/2611 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11222 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/16694 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/11070 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/14546 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/12842 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebDriver/Session.cpp:
```diff
@@ -3211,13 +3212,22 @@ void Session::doLogEntryAdded(RefPtr<JSON::Object>&& message)
+        entry->setString("method"_s, method);
+        entry->setArray("args"_s, args);
+    } else if (messageType == "javascript"_s) {
+        // Despite not having the stack information, the spec requires this
+        // field for JavaScript messages.
+        entry->setString("stackTrace"_s, emptyString());
```
the `ConsoleMessage` has this data.  can we not send it as part of `Automation.logEntryAdded`?

lauromoura replied:
Yes, the idea is to do this in a follow-up bug (https://bugs.webkit.org/show_bug.cgi?id=282979), translating it to the format defined by the BiDi spec.

dcrousso commented on Source/WebDriver/Session.cpp:
```diff
@@ -3211,13 +3212,22 @@ void Session::doLogEntryAdded(RefPtr<JSON::Object>&& message)
+
+    if (messageType == "console"_s) {
+        entry->setString("method"_s, method);
+        entry->setArray("args"_s, args);
```
can we move the code that create `args` to inside this `if` so it doesnt have to happen for _every_ message?

or will `args` be used by other `messageType` too?

lauromoura replied:
Good catch. `args` is just used in `console` messages, as part of the `log.ConsoleLogEntry` rule in https://w3c.github.io/webdriver-bidi/#types-log-logentry.

I've moved it inside the `if` block and will merge after one last EWS run. Thanks for the review.

lauromoura committed:
[WebDriver][BiDi] Use "javascript" as the "type" parameter when the message comes from a JavaScript error
https://bugs.webkit.org/show_bug.cgi?id=285028

Reviewed by Devin Rousso.

Use the `messageSource` field to check if the message is coming from a
JS error or from a regular console method.

Also set the `stackTrace` field for JS messages, as the spec requires
it.

* Source/WebDriver/Session.cpp:
(WebDriver::Session::doLogEntryAdded):
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::logEntryTypeForMessage):
(WebKit::WebAutomationSession::logEntryAdded):

Canonical link: https://commits.webkit.org/288666@main
